perl: error: EXTERN.h: No such file or directory

Hello, when I tried to upgrade `perl` in CentOS 6, I met
```
stacksize: Perl lib version (5.28.0) doesn't match executable '/home/skpark/tmp/perl-20180730-25515-15ld1j5/perl-5.28.0/perl' version (5.26.2) at ../../lib/Config.pm line 62. 
Compilation failed in require at stacksize line 5.
BEGIN failed--compilation aborted at stacksize line 5.
make[1]: *** [lib/Storable/Limit.pm] Error 255
```

Thank you.
[https://gist.github.com/a1d1a1ca62a853ff506340e395a4ff8a](https://gist.github.com/a1d1a1ca62a853ff506340e395a4ff8a)

- [x] have a problem with `brew install` (or `upgrade`, `reinstall`) a single formula? If it's a general `brew` problem please file this issue at Linuxbrew/brew: https://github.com/Linuxbrew/brew/issues/new/choose. If it's a tap (e.g. Brewsci/homebrew-bio) problem please file this issue at the tap.
- [x] ran `brew update` and can still reproduce the problem?
- [x] ran `brew doctor`, fixed all issues and can still reproduce the problem?
- [x] ran `brew gist-logs <formula>` (where `<formula>` is the name of the formula that failed) and included the output link?
- [x] if `brew gist-logs` didn't work: ran `brew config` and `brew doctor` and included their output with your issue?

I met another error message:
```
Expat.xs:14:20: fatal error: EXTERN.h: No such file or directory
compilation terminated.
make[1]: *** [Expat.o] Error 1
```

But, I added
```
87         ENV["C_INCLUDE_PATH"] = Formula["perl"].lib/"perl5/#{version}/x86_64-linux-thread-multi/CORE"
```
into perl.rb. The problem was solved. I will try to submit my RB.

Perl now successfully installs for me, but the problem with texinfo remains:

```
libtool: compile:  gcc -DHAVE_CONFIG_H -I. -I. -I./lib -I./lib -D_REENTRANT -D_GNU_SOURCE -fno-strict-aliasing -pipe -fstack-protector -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -DVERSION=\"0\" -DXS_VERSION=\"1\" -I/usr/lib64/perl5/CORE -c XSParagraph.c  -fPIC -DPIC -o .libs/XSParagraph_la-XSParagraph.o
xspara.c:39:20: fatal error: EXTERN.h: No such file or directory
```
The gist is here:
https://gist.github.com/definelicht/c7c9207fdf1810b254c11381fd9244fc

--- a/Formula/texinfo.rb
+++ b/Formula/texinfo.rb
@@ -23,6 +23,7 @@ class Texinfo < Formula
   def install
     system "./configure", "--disable-dependency-tracking",
                           "--disable-install-warnings",
+                          *("--disable-perl-xs" unless OS.mac?),
                           "--prefix=#{prefix}"
     system "make", "install"
     doc.install Dir["doc/refcard/txirefcard*"]

